@Joffref’s WS setup improved

Nedomas opened the pull request:
Hi @Joffref!

First of all, amazing job with the WS gateway, works great as far as I tested. Most of the code here is just like your PR’, just improved a few small things + split everything into multiple files (since it started to get less manageable).

What’s changed/done in addition to your WS stuff:
1. Instead of `--ws` flag, introduced `--outputTransport ws` flag. It will help as we introduce more transports. By default now, whatever was working before will keep on working, so we don’t break the api. So if people do `--stdio "something"` without any other flags, `--outputTransport` will auto-default to `sse` and visa versa. 
2. Regarding `healthPort` - just simplified it to use `--port` instead of additional `--healthPort` flag, but if you had a specific use-case where in WS `--port` and `--healthPort` must differ and it’s very legit, then we should support this use-case. I’d suggest maybe we could reuse `--healthEndpoint` then and just allow something like `--healthEndpoint /healthz:8000`.
3. Split `stdioToSse`, `sseToStdio`, `stdioToWs` into multiple files as it was getting bad.
4. Updated readme.
5. Added prettier - `websocket.ts` is not changed at all! It is just prettier reformatting, but there is not change to `websocket.ts` in this PR and it’s exactly as you left it.

Would love your review and if you agree to these changes ^

I’ve tested this and it seems to be working as it was, so the changes here are mostly cosmetics around CLI api.

If everything is alright, I’d release this.

Joffref committed:
Draft implementation of WebSockets - Not yet tested

Signed-off-by: Joffref <mjoffre@beamlit.com>

Nedomas committed:
Update example SSE url to Supermachine-hosted SQLite

Nedomas committed:
v2.3.1: handle exit signals better

Nedomas committed:
v2.3.2: improve exit signals for Claude Desktop and Docker

Nedomas committed:
Add official docker image; update README

Joffref committed:
Enhance WebSocket support: Add CORS and health endpoint configuration options. Update README to reflect new command-line options for WebSocket and health checks.

Signed-off-by: Joffref <mjoffre@beamlit.com>

Nedomas committed:
Merge remote-tracking branch 'origin/main' into joffref-ws-updated-fork

Nedomas committed:
Split into gateways; add --outputTransport

Nedomas committed:
Extract to onSignals

Nedomas committed:
Add prettier

Nedomas committed:
Add husky, lint-staged

Nedomas committed:
Remove pnpm-lock

Nedomas committed:
Simplify healthPort and improve README

Nedomas committed:
Just identical response

Nedomas committed:
Add stdio → WS to readme, organize a bit better

Nedomas committed:
Cleaner logs

Joffref commented on package.json:
```diff
@@ -20,21 +20,33 @@
     "express": "^4.21.2",
+    "uuid": "^11.1.0",
+    "ws": "^8.18.1",
     "yargs": "^17.7.2"
```
Zod is missing here

Joffref commented on src/gateways/stdioToWs.ts:
```diff
@@ -0,0 +1,158 @@
+        }
+      })
+    }
+    app.listen(port, hostname, () => {
```
It's not possible to use the same port since it fails when the transport starts because port is already taken.

Nedomas replied:
Gotcha, should have tested this more 😅

Joffref replied:
No worries! Can't wait to see it merged!

Joffref reviewed (commented):
Overall a good improvement of the first PR, you did an amazing job and yes it's clearer.

Nedomas commented:
Thanks @Joffref , will clean up the healthEndpoints then and other mentions and will merge it all

michaeljguarino committed:
Setup Docker image build

this will set up builds on both pr and docker tag, for both amd64 and arm64 architectures

Nedomas committed:
Add dockerhub

Nedomas committed:
Merge pull request #22 from supercorp-ai/feature/github-actions-docker-build

GitHub actions docker build / refs @michaeljguarino

Nedomas committed:
Add contributors

Nedomas committed:
Add ghcr image

Nedomas committed:
Add health endpoints and cors headers on express server with ws

Nedomas committed:
Run websocket and health endpoints on same express server

Nedomas committed:
Merge remote-tracking branch 'origin/main' into joffref-ws-updated-fork

Nedomas committed:
Improve readme

Nedomas committed:
Move port a bit down

Nedomas committed:
Update example urls; @Joffref mention

Nedomas committed:
Update url in code demo

Nedomas committed:
Update support

Nedomas committed:
Update readme

Nedomas commented:
Hey, this was just closed and released in supergateway@2.4.0: https://github.com/supercorp-ai/supergateway/pull/10

Thanks!
